src/crimson/os/seastore/linked_tree_node.h

```diff
@@ -961,39 +974,25 @@ class ParentNode {
 template <typename ParentT, typename T, typename key_t>
 class ChildNode : public BaseChildNode<ParentT, key_t> {
 protected:
-  void take_prior_parent_tracker() {
-    auto &me = down_cast();
-    auto &prior = static_cast<T&>(*me.get_prior_instance());
-    this->parent_tracker = prior.parent_tracker;
-  }
-
-  void set_parent_tracker_from_prior_instance() {
-    auto &me = down_cast();
-    assert(!me.is_btree_root());
-    assert(me.is_mutation_pending());
-    take_prior_parent_tracker();
-    assert(me.is_parent_valid());
-    auto parent = this->get_parent_node();
-    //TODO: can this search be avoided?
-    auto off = parent->lower_bound(me.get_begin()).get_offset();
-    assert(parent->iter_idx(off).get_key() == me.get_begin());
-    parent->children[off] = &me;
-  }
-
   void on_invalidated() {
     this->reset_parent_tracker();
   }
+  void take_parent_from_prior() {
+    _take_parent_from_prior();
+  }
   void on_replace_prior() {
-    set_parent_tracker_from_prior_instance();
+    take_parent_from_prior();
   }
   void destroy() {
-    auto &me = down_cast();
-    assert(!me.is_btree_root());
-    ceph_assert(me.is_parent_valid());
+    assert(!down_cast().is_btree_root());
+    if (!this->has_parent_tracker()) {
+      // XXX: On startup, parent/child nodes may not be linked,
+      //      we need to tolerate this case.
+      return;
+    }
```

Comment:
Grepping the prior users of `destroy()`, can this be consistent?

For example, move `has_parent_tracker()` to the callers of `destory()` in this PR.

Or keep this version and remove `has_parent_tracker()` checks from the callers in the lba paths.

Reply:
I'll try that:-)

Reply:
Um, it seems the callers of `destroy()` don't call `has_parent_tracker()`, am I missing something?

Reply:
I see `~LogicalChildNode()` is also checking `has_parent_tracker()`.

Why `~FixedKVInternalNode()` and `~FixedKVLeafNode()` was working without checking `has_parent_tracker()`? What's the key difference ?

Reply:
> I see `~LogicalChildNode()` is also checking `has_parent_tracker()`.

Sorry, this should be a mistake, will remove it:-)

Reply:
Done:-)

Reply:
@xxhdx1985126

Why ~FixedKVInternalNode() and ~FixedKVLeafNode() was working without checking `has_parent_tracker()`?

Or they worked because the corner case wasn't hit before?

Reply:
I think it shouldn't happen, because a valid `ChildNode` should always be pointing to a valid parent through the parent tracker in the lba tree. Will add assertions to make sure this.